### ISSUE TITLE
Semantic versioning constraint in 'helm search' doesn't work without --versions

### ISSUE THREAD

**Issue by mparry** [kubernetes/helm#3115]
It will only ever report the latest version, or nothing at all if that doesn't satisfy the constraint. For example, if `my-chart` exists at versions 1.0, 1.1, 1.2, and 2.0 then:

```
> helm search --version "> 1, < 2" my-chart
No results found
```

This appears to be because the calls to `i.AddRepo()` in `searchCmd.buildIndex()` do not set the `all` flag, so the constraint is only ever applied against the latest version of each matching chart.

Note that it _does_ apply the constraint if `--versions` is also specified, eg.

```
> helm search --versions --version "> 1, < 2" my-chart
NAME      VERSION  DESCRIPTION
my-chart   1.2.0    ...
my-chart   1.1.0    ...
```

but it doesn't seem like this should be necessary, and the help text doesn't imply this.

In summary, specifying `--version` alone should give the single (latest) matching version.

**Comment by mparry** [comment]
I have a change (PR coming shortly) that will make it behave like this instead:

```
> helm search --versions my-chart
NAME      VERSION  DESCRIPTION
my-chart   2.0.0    ...
my-chart   1.2.0    ...
my-chart   1.1.0    ...
my-chart   1.0.0    ...
                                                                          
> helm search --versions --version "> 1, < 2" my-chart
NAME      VERSION  DESCRIPTION
my-chart   1.2.0    ...
my-chart   1.1.0    ...    
                                                                          
> helm search --version "> 1, < 2" my-chart             
NAME      VERSION  DESCRIPTION
my-chart   1.2.0    ...

> helm search app-comet-explorer
NAME      VERSION  DESCRIPTION
my-chart   2.0.0    ...
```
